src/setup/validateTxnFields.ts

```diff
@@ -199,12 +202,19 @@ export const validateTxnFields =
         }
         debugAppdata.set(txHash, appData)
 
-        if (isSigned && isSignatureValid) {
+        if (!gasValid) {
+          success = false
```

Comment:
success is only being set to false in !gasValid but not in the other two checks

Reply:
resolved with early returns :grin: